src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,992 @@ static ColliderJntSphInit D_80AD2790 = {
+        if (this->actor.colChkInfo.damageEffect == DMG_EFF_FIRE) {
+            Vec3f pos;
+            s32 i;
+            for (i = 4; i >= 0; i--) {
+                pos.x = Rand_CenteredFloat(20.0f) + this->actor.world.pos.x;
+                pos.y = (Rand_ZeroOne() * 25.0f) + this->actor.world.pos.y;
+                pos.z = Rand_CenteredFloat(20.0f) + this->actor.world.pos.z;
+                EffectSsEnFire_SpawnVec3f(globalCtx, &this->actor, &pos, 70, 0, 0, -1);
+            }
+            func_8003426C(&this->actor, 0x4000, 200, 0, 100);
+        }
+        if (this->actor.colChkInfo.health == 0) {
+            EnPeehat_Adult_SetStateDie(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD167C.s")
+void EnPeehat_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+    s32 i;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD1700.s")
+    // If Adult Peahat
+    if (thisx->params <= 0) {
+        EnPeehat_Adult_CollisionCheck(this, globalCtx);
+    }
+    if (thisx->colChkInfo.damageEffect != DMG_EFF_LIGHT_ICE_ARROW) {
+        if (thisx->speedXZ != 0.0f || thisx->velocity.y != 0.0f) {
+            Actor_MoveForward(thisx);
+            Actor_UpdateBgCheckInfo(globalCtx, thisx, 25.0f, 30.0f, 30.0f, 5);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80AD177C.s")
+        this->actionFunc(this, globalCtx);
+        if ((globalCtx->gameplayFrames & 0x7F) == 0) {
+            this->jiggleRotInc = (Rand_ZeroOne() * 0.25f) + 0.5f;
+        }
+        this->jiggleRot += this->jiggleRotInc;
+    }
+    // if TYPE_GROUNDED
+    if (thisx->params < 0) {
+        // Set the Z-Target point on the Peahat's weak point
+        thisx->focus.pos.x = this->colJntSph.elements[0].dim.worldSphere.center.x;
+        thisx->focus.pos.y = this->colJntSph.elements[0].dim.worldSphere.center.y;
+        thisx->focus.pos.z = this->colJntSph.elements[0].dim.worldSphere.center.z;
+        if (this->state == PEAHAT_STATE_SEEK_PLAYER) {
+            Math_SmoothStepToS(&thisx->shape.rot.x, 6000, 1, 300, 0);
+        } else {
+            Math_SmoothStepToS(&thisx->shape.rot.x, 0, 1, 300, 0);
+        }
+    } else {
+        thisx->focus.pos = thisx->world.pos;
+    }
+    Collider_UpdateCylinder(thisx, &this->colCylinder);
+    if (thisx->colChkInfo.health > 0) {
+        // If Adult Peahat
+        if (thisx->params <= 0) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colCylinder.base);
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colJntSph.base);
+            if (thisx->colorFilterTimer == 0 || (thisx->colorFilterParams & 0x4000) == 0) {
+                if (this->state != PEAHAT_STATE_EXPLODE) {
+                    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colJntSph.base);
+                }
```

Comment:
```suggestion
            if ((thisx->colorFilterTimer == 0 || !(thisx->colorFilterParams & 0x4000)) && (this->state != PEAHAT_STATE_EXPLODE)) {
                CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colJntSph.base);
```
Combine these conditionals?